### DISCUSSION TITLE
What do we need to do to prepare for an interview as a Junior Software Engineer?

### DISCUSSION THREAD

**Discussion by solaris5** [MomoSolaris8/superPower#4] [Q&A] [answered]
what we do as a  recent / upcoming CS graduate ,in an entry level software developer job?

**Comment by MomoSolaris8** [comment, marked as answer]
In CS interview books, they tell you that you need to know data structures and algorithms, and be able to solve problems for the interview.  It is true, you need to figure out what difference between university and industrial world (the real word).

1.  Of course, you have to learn some Algos like Blind 75 or LeetCode .
2. A lot of work is building CRUD apps. This may be what you're interviewing for and if so, hopefully the interviewers will not be asking you to reverse a linked list or some garbage like that.
Knowledge of source control, unit testing, integration testing and general familiarity with working on a team are some things I would ask about.

**Reply by solaris5** [reply]
Since IT/software junior developers are not easy or difficult, most companies are looking for developers with 1+/2+ years of experience. However, how does a pre-bachelor CS student get experience? You need 1/2 working student work, i.e. you actually do a CODING working student, not an IT management working student.

**Reply by MomoSolaris8** [reply]
Not necessarily, in fact, do not need to have a lot of experience or no experience at all can. This will only affect the salary of your first junior SE, the new junior SE to the company still needs training, but also to relearn the new framework. This is not a problem
